packages/wallet/src/services/BalanceTracker.ts

```diff
@@ -1,28 +1,52 @@
-import { Balance, TransactionalTracker } from './types';
-import { Cardano } from '@cardano-sdk/core';
+import { Balance, Delegation, DelegationKeyStatus, TransactionalObservables, TransactionalTracker } from './types';
+import { Cardano, ProtocolParametersRequiredByWallet } from '@cardano-sdk/core';
+import { Observable, combineLatest, distinctUntilChanged, map, share } from 'rxjs';
 import { TrackerSubject } from './util';
-import { TransactionalObservables } from '..';
-import { combineLatest, map } from 'rxjs';
 
-// TODO: subtract deposit quantity from total utxo coin as it can't be spent
-// Review: not sure how to represent this. Is it 'balance.available$'? 'balance.total$'? a new one?
-// 'total' represents total value available to spend. Would be confusing if it includes deposit.
-// 'available' represents value available to spend if it makes a transaction right now:
-// would be confusing if it's never equal to total after stake registration
-const mapToBalances = map<[Cardano.Utxo[], Cardano.Lovelace], Balance>(([utxo, rewards]) => ({
-  ...Cardano.util.coalesceValueQuantities(utxo.map(([_, txOut]) => txOut.value)),
-  rewards
-}));
+const mapToBalances = map<[Cardano.Utxo[], Cardano.Lovelace, Cardano.Lovelace], Balance>(
+  ([utxo, rewards, deposit]) => ({
+    ...Cardano.util.coalesceValueQuantities(utxo.map(([_, txOut]) => txOut.value)),
+    deposit,
+    rewards
+  })
+);
+
+const createDepositTracker = (
+  protocolParameters$: Observable<ProtocolParametersRequiredByWallet>,
+  numDeposits$: Observable<number>
+) =>
+  combineLatest([numDeposits$, protocolParameters$]).pipe(
+    map(([registeredAccounts, { stakeKeyDeposit }]) => BigInt(registeredAccounts * stakeKeyDeposit)),
+    distinctUntilChanged()
+  );
+
+const numRewardAccountsWithKeyStatus = (delegationTracker: Delegation, keyStatus: DelegationKeyStatus) =>
+  delegationTracker.rewardAccounts$.pipe(
+    map((accounts) => accounts.filter((account) => account.keyStatus === keyStatus).length)
+  );
 
 export const createBalanceTracker = (
+  protocolParameters$: Observable<ProtocolParametersRequiredByWallet>,
   utxoTracker: TransactionalObservables<Cardano.Utxo[]>,
-  rewardsTracker: TransactionalObservables<Cardano.Lovelace>
+  rewardsTracker: TransactionalObservables<Cardano.Lovelace>,
+  delegationTracker: Delegation
 ): TransactionalTracker<Balance> => {
+  const depositTotal$ = createDepositTracker(
```

Comment:
Yep, this is a good representation. It uses the domain terminology to make an explicit categorisation of the unusable funds